### PR TITLE
added blog files

### LINKED ISSUE #582: [General] Add few blogs related to algorithms
### Issue Description
Please describe the issue or question in detail.

Common Recursion Patterns: Discuss various recursion patterns (e.g., tree traversal, backtracking) with code examples.
Recursion vs. Iteration: A comparative analysis of recursion and iteration, including pros and cons.
Practical Applications of Recursion: Explore real-world applications of recursion in algorithms, such as sorting or searching.
Optimizing Recursive Functions: Tips and techniques for optimizing recursive algorithms (e.g., memoization, tail recursion).
Understanding Big O Notation: A deep dive into time and space complexity, including examples and visual representations.

### Additional Context
Add any other context, code snippets, or screenshots related to the issue or question.

---

**Would you like to work on this issue?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by AKSHITHA-CHILUKA**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

added all the blog files related to 
Common Recursion Patterns
Recursion vs. Iteration
Practical Applications of Recursion
Optimizing Recursive Functions
Understanding Big O Notation

Fixes #582 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: blog/Recursion-vs-Iteration.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+
+### Blog: Recursion vs. Iteration
```
Comment (ajay-dhangar) [inline blog/Recursion-vs-Iteration.md]: delete your line number 2 `### Blog: Recursion vs. Iteration`

File: blog/Common-Recursion-Patterns.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Blog : common-recursion-patterns
```
Comment (ajay-dhangar) [inline blog/Common-Recursion-Patterns.md]: delete line number 1

File: blog/optimizing-recursive-functions.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+
+### Blog 4: Optimizing Recursive Functions
```
Comment (ajay-dhangar) [inline blog/optimizing-recursive-functions.md]: delete line number 2

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Before writing blog content, start by adding your basic information to the `authors.yml` file. You can refer to the example here: [authors.yml](https://github.com/ajay-dhangar/algo/blob/853d76632fb0020059f96dcf2430ba34baafbfba/blog/authors.yml#L18). Once that's done, make sure to follow the standard format for your blog post:

```md
---
slug: mastering-recursion
title: 'Mastering Recursion: Concepts, Problems, and Optimization'
authors: [narendra-dhangar]
tags: [recursion, algorithms, dsa, optimization, narendra-dhangar]
---
```

**Review by AKSHITHA-CHILUKA — COMMENTED** [review]
added the details in the author.yml file and deleted the first lines.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**